thonny/editors.py

```diff
@@ -92,8 +92,15 @@ def update_file_type(self):
         if self._filename is None:
             self._code_view.set_file_type(None)
         else:
-            ext = self._filename.split(".")[-1].lower()
-            if ext in PYTHON_EXTENSIONS:
+            # Include backup files: *.py.bak, *.py.orig, etc
+            get_dots = len(re.findall(r'\.', self._filename))
+            ext = self._filename.split(".")[-get_dots].lower()
```

Comment:
If the filename is "blah.blah.py", the extension would be "blah"? This doesn't look right.